src/cpu/x64/jit_brgemm_weights_decompression_kernel.cpp

```diff
@@ -51,6 +51,17 @@ void jit_brgemm_weights_decompression_kernel_t<isa>::init_decomp_params(std::fun
                     uni_vbroadcastss(vmm_params(ocb), xmm_params);
                     break;
                 }
+                case data_type::u2: {
+                    auto xmm_params = Xmm(vmm_params(ocb).getIdx());
+                    auto reg_tmp_32 = Reg32(reg_tmp.getIdx());
+                    movzx(reg_tmp_32, ptr[reg_params]);
+                    uni_vmovq(xmm_params, reg_tmp);
+                    uni_vpslld(xmm_params, xmm_params, 30);
+                    uni_vpsrld(xmm_params, xmm_params, 30);
+                    uni_vcvtdq2ps(xmm_params, xmm_params);
+                    uni_vbroadcastss(vmm_params(ocb), xmm_params);
```

Comment:
```suggestion
                    and_(reg_tmp_32, 0x3);
                    uni_vmovd(xmm_params, reg_tmp);
                    uni_vcvtdq2ps(xmm_params, xmm_params);
                    uni_vbroadcastss(vmm_params(ocb), xmm_params);
```
In this  particular case we can mask bits using GPR.

Reply:
Good point! Applied. Thanks Maksim!